dask/array/reductions.py

```diff
@@ -223,7 +235,23 @@ def partial_reduce(func, x, split_every, keepdims=False, dtype=None, name=None):
         g = lol_tuples((x.name,), range(x.ndim), decided, dummy)
         dsk[(name,) + k] = (func, g)
     graph = HighLevelGraph.from_collections(name, dsk, dependencies=[x])
-    return Array(graph, name, out_chunks, dtype=dtype)
+
+    meta = x._meta
+    if reduced_meta is not None:
+        try:
+            meta = func(reduced_meta.compute(scheduler=core.get), meta=True)
```

Comment:
@mrocklin as per our offline conversation earlier today, you mentioned using this scheduler would not trigger any computation on workers, but instead run locally on the client side. Since `._meta` has all dimensions of length 0, the compute power here will be mostly irrelevant.

IMO, this is a better solution than not reducing `._meta`, as it will have potential for handling more complex types, and that wouldn't necessarily be the case by doing a straightforward computation of `._meta` on `reduction()`. I also spent sometime earlier today trying to avoid this process, while it's possible for some cases, it wouldn't simplify the code complexity too much.

Given that, my vote is to do the reduction, as long as we can do it all on the client side, which was indeed my expectation all along, unfortunately, I was ignorant of the differences on all schedulers.